src/sentry/static/sentry/app/views/projectDetail/charts/projectApdexChart.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import * as ReactRouter from 'react-router';
+import * as Sentry from '@sentry/react';
+import {Location} from 'history';
+
+import {fetchTotalCount} from 'app/actionCreators/events';
+import {Client} from 'app/api';
+import EventsChart from 'app/components/charts/eventsChart';
+import QuestionTooltip from 'app/components/questionTooltip';
+import {t} from 'app/locale';
+import {GlobalSelection, Organization} from 'app/types';
+import {getUtcDateString} from 'app/utils/dates';
+import EventView, {isAPIPayloadSimilar} from 'app/utils/discover/eventView';
+import getDynamicText from 'app/utils/getDynamicText';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import {getTermHelp} from 'app/views/performance/data';
+import {HeaderTitleLegend} from 'app/views/performance/styles';
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  location: Location;
+  router: ReactRouter.InjectedRouter;
+  selection: GlobalSelection;
+  onTotalValuesChange: (value: number | null) => void;
+};
+
+class ProjectApdexChart extends React.Component<Props> {
+  componentDidMount() {
+    this.fetchTotalCount();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const currentAPIPayload = this.getEventView(this.props).getEventsAPIPayload(
+      this.props.location
+    );
+    const previousAPIPayload = this.getEventView(prevProps).getEventsAPIPayload(
+      prevProps.location
+    );
```

Comment:
Do you need an EventView here given that most of it is static? You could compare the global selection properties to detect if you need to reload.